build_rules.gradle

```diff
@@ -344,7 +344,7 @@ class JavaNatureConfiguration {
   /** Controls whether the errorprone plugin is enabled and configured */
   boolean enableErrorProne = true
   /** Controls whether compiler warnings are treated as errors */
-  boolean failOnWarning = false
```

Comment:
My preference is to remove this configuration altogether and make the `failOnWarning` behavior default.

Reply:
We can't remove this for the moment we have modules still skipping using gthis e.g. 'sdks/java/core'.

Reply:
Got it. I'll work on finishing up `sdks/java/core` as I have time and then remove this. /cc @pabloem